tests/fdb/test_fdb.py

```diff
@@ -160,12 +160,24 @@ def send_recv_eth(ptfadapter, source_ports, source_mac, dest_ports, dest_mac, sr
         exp_pkt.set_do_not_care_scapy(scapy.Dot1Q, "prio")
     logger.debug('send packet src port {} smac: {} dmac: {} vlan: {} verifying on dst port {}'.format(
         source_ports, source_mac, dest_mac, src_vlan, dest_ports))
-    ptfadapter.dataplane.flush()
-    testutils.send(ptfadapter, source_ports[0], pkt)
-    if len(dest_ports) == 1:
-        testutils.verify_packet(ptfadapter, exp_pkt, dest_ports[0], timeout=FDB_WAIT_EXPECTED_PACKET_TIMEOUT)
+
+    # fdb test will send lots of pkts between paired ports, it's hard to guarantee there is no congestion
+    # on server side during this period. So tolerant to retry 3 times before complain the assert.
+
+    retry_count = 3
+    for _ in range(retry_count):
+        try:
+            ptfadapter.dataplane.flush()
+            testutils.send(ptfadapter, source_ports[0], pkt)
+            if len(dest_ports) == 1:
+                testutils.verify_packet(ptfadapter, exp_pkt, dest_ports[0], timeout=FDB_WAIT_EXPECTED_PACKET_TIMEOUT)
+            else:
+                testutils.verify_packet_any_port(ptfadapter, exp_pkt, dest_ports, timeout=FDB_WAIT_EXPECTED_PACKET_TIMEOUT)
+            break
+        except:
+            pass
     else:
-        testutils.verify_packet_any_port(ptfadapter, exp_pkt, dest_ports, timeout=FDB_WAIT_EXPECTED_PACKET_TIMEOUT)
+        pytest_assert(False, "Expected packet was not received on ports {}".format(dest_ports))
```

Comment:
Can you please dump the fdb from the DUT with `fdb show` to confirm if the FDB entry is present when test is failing? That will help us debug when the test fails again.

Reply:
Sure